token/program-2022-test/tests/reallocate.rs

```diff
@@ -188,3 +195,93 @@ async fn reallocate_without_current_extension_knowledge() {
         ])
     );
 }
+
+#[test_case(&[ExtensionType::CpiGuard], true, true ; "transfer lamports and sync with extension")]
+#[test_case(&[ExtensionType::CpiGuard], true, false ; "transfer lamports only with extension")]
+#[test_case(&[ExtensionType::CpiGuard], false, false ; "do not transfer, with extension")]
+#[test_case(&[], true, true ; "transfer lamports and sync without extension")]
+#[test_case(&[], true, false ; "transfer lamports only without extension")]
+#[test_case(&[], false, false ; "do not transfer, without extension")]
+#[tokio::test]
+async fn reallocate_updates_native_rent_exemption(
+    extensions: &[ExtensionType],
+    transfer_lamports: bool,
+    sync_native: bool,
+) {
+    let mut context = TestContext::new().await;
+    context.init_token_with_native_mint().await.unwrap();
+    let TokenContext { token, alice, .. } = context.token_context.unwrap();
+    let context = context.context.clone();
+
+    let alice_account = Keypair::new();
+    token
+        .create_auxiliary_token_account(&alice_account, &alice.pubkey())
+        .await
+        .unwrap();
+    let alice_account = alice_account.pubkey();
+
+    // transfer more lamports
+    let transfer_amount = 1_000_000_000;
```

Comment:
What did you think about ensuring the `ExtensionType::CpiGuard` cases still work when 0 < this amount < 41760?

Reply:
oh yes, sorry, i'll add that too